File: okj/rendszeruzemelteto/koridok/rust/AutoVerseny.rs
use std::{ fs::File, io::BufReader, io::{ BufRead, stdin } };
use chrono::{ NaiveTime, Timelike };
use itertools::Itertools;

fn main() {
    let input_file = File::open("autoverseny.csv").unwrap();
    let versenyek = BufReader::new(input_file)
                             .lines()
                             .skip(1)
                             .map(|k| create_verseny(&k.unwrap()))
                             .collect::<Vec<_>>();

    println!("3. Feladat: Adatsorok száma: {}", versenyek.len());

    match versenyek.iter().find(|k| k.versenyzo == "Fürge Ferenc" && k.palya == "Gran Prix Circuit" && k.kor == 3) {
        None => {},
        Some(k) => println!("4. Feladat: {} mp", k.korido.num_seconds_from_midnight())
    }

    println!("5. Felatad: Írj be egy nevet!");

    let mut be_nev_str = String::new();
    stdin().read_line(&mut be_nev_str);

    let be_nev = be_nev_str.trim_end();

    match versenyek.iter().filter(|k| k.versenyzo == be_nev).min_by_key(|k| k.korido) {
        None => println!("6. Feladat: Nincs ilyen versenyző"),
        Some(k) => println!("6. Feladat: {}", k.korido)
    }
}

fn create_verseny(line: &String) -> Verseny {
    let split = line.split(';').collect::<Vec<_>>();

    Verseny {
        csapat: split[0].to_string(),
        versenyzo: split[1].to_string(),
        eletkor: split[2].parse::<i32>().unwrap(),
        palya: split[3].to_string(),
        korido: NaiveTime::parse_from_str(split[4], "%H:%M:%S").unwrap(),
        kor: split[5].parse::<i32>().unwrap()
    }
}

struct Verseny {
    csapat: String,
    versenyzo: String,
    eletkor: i32,
    palya: String,
    korido: NaiveTime,
    kor: i32
}

File: okj/rendszeruzemelteto/koridok/python/AutoVerseny.py
from dataclasses import dataclass
from datetime import time

@dataclass
class Verseny:
    csapat: str
    versenyzo: str
    eletkor: int
    palya: str
    korido: time
    kor: int


with open('autoverseny.csv', encoding = 'utf-8') as input_file:
    create_verseny = lambda split: Verseny(split[0], split[1], int(split[2]), split[3], time.fromisoformat(split[4]), int(split[5]))
    versenyek = [ create_verseny(k.split(';')) for k in input_file.readlines()[1:] ]

print(f'3. Feladat: Adatsorok száma: {len(versenyek)}')

kivalasztott = next(k for k in versenyek if k.versenyzo == 'Fürge Ferenc' and k.palya == 'Gran Prix Circuit' and k.kor == 3).korido
print(f'4. Feladat: {(kivalasztott.hour * 60 + kivalasztott.minute) * 60 + kivalasztott.second} mp')

be_nev = input('5. Felatad: Írj be egy nevet: ')
be_versenyzo_korido = min((k.korido for k in versenyek if k.versenyzo == be_nev), default = None)

print(f'6. Feladat: {be_versenyzo_korido}' if be_versenyzo_korido != None else '6. Feladat: Nincs ilyen versenyző')